src/base/utils.js

```diff
@@ -235,10 +235,34 @@ export function removeArrayItem(arr, item) {
   }
 }
 
+// HTML5 video element recycle storage object.
+// If playback recycleVideo flag is set, it store
+// one video element for each player instance created.
+const videoStore = {}
+
+export class VideoStore {
```

Comment:
Another suggestion I'd like to do is to create a stack of `video` elements instead of using `playerId` as a key since we can have more than one `video` element in a single player (for example, if you have more than one container).